Training5/task1/src/features/todos/todosSlice.js

```diff
@@ -0,0 +1,82 @@
+import { createSlice, createSelector } from '@reduxjs/toolkit'
+import { StatusFilters } from "../filters/filtersSlice";
+import axios from 'axios'
+
+const initialState = []
+
+const todosSlice = createSlice({
+    name: 'todos',
+    initialState,
+    reducers: {
+        todosAdded(state, action) {
+            return [...state, action.payload]
+        },
+        todosToggled(state, action) {
+            const todo = state.find(todo => todo.id === action.payload)
+            todo.completed = !todo.completed
+        },
+        todosDeleted(state, action) {
+            return state.filter(todo => todo.id !== action.payload)
+        },
+        todosSearch(state, action) {
```

Comment:
@Doan-Khanh  We should not filter todos when the user changes the search, because you can't revert back if the search is empty, because it should search based on fetched todos, not based on filtered todos